src/amo/reducers/addonsByAuthors.js

```diff
@@ -1,123 +1,136 @@
-  }
+export const fetchAddonsByAuthors = (
+  { addonType, authors, errorHandlerId, excludeAddonBySlug }: FetchAddonsByAuthorsParams
+): FetchAddonsByAuthorsAction => {
+  invariant(errorHandlerId, 'An errorHandlerId is required');
+  invariant(addonType, 'An add-on type is required.');
+  invariant(authors, 'Authors are required.');
+  invariant(Array.isArray(authors), 'The authors parameter must be an array.');
 
   return {
-    type: FETCH_OTHER_ADDONS_BY_AUTHORS,
+    type: FETCH_ADDONS_BY_AUTHORS,
     payload: {
       addonType,
       authors,
       errorHandlerId,
-      slug,
+      excludeAddonBySlug,
     },
   };
 };
 
-type LoadOtherAddonsByAuthorsParams = {|
-  slug: string,
+type LoadAddonsByAuthorsParams = {|
   addons: Array<ExternalAddonType>,
+  excludeAddonBySlug?: string,
 |};
 
-type LoadOtherAddonsByAuthorsAction = {|
-  type: typeof LOAD_OTHER_ADDONS_BY_AUTHORS,
-  payload: LoadOtherAddonsByAuthorsParams,
+type LoadAddonsByAuthorsAction = {|
+  type: typeof LOAD_ADDONS_BY_AUTHORS,
+  payload: LoadAddonsByAuthorsParams,
 |};
 
-export const loadOtherAddonsByAuthors = (
-  { addons, slug }: LoadOtherAddonsByAuthorsParams
-): LoadOtherAddonsByAuthorsAction => {
-  if (!slug) {
-    throw new Error('An add-on slug is required.');
-  }
-
-  if (!addons) {
-    throw new Error('A set of add-ons is required.');
-  }
+export const loadAddonsByAuthors = (
+  { addons, excludeAddonBySlug }: LoadAddonsByAuthorsParams
+): LoadAddonsByAuthorsAction => {
+  invariant(addons, 'A set of add-ons is required.');
 
   return {
-    type: LOAD_OTHER_ADDONS_BY_AUTHORS,
-    payload: { slug, addons },
+    type: LOAD_ADDONS_BY_AUTHORS,
+    payload: { addons, excludeAddonBySlug },
   };
 };
 
 type Action =
-  | FetchOtherAddonsByAuthorsAction
-  | LoadOtherAddonsByAuthorsAction;
+  | FetchAddonsByAuthorsAction
+  | LoadAddonsByAuthorsAction;
 
 const reducer = (
   state: State = initialState,
   action: Action
 ): State => {
   switch (action.type) {
-    case FETCH_OTHER_ADDONS_BY_AUTHORS:
+    case FETCH_ADDONS_BY_AUTHORS:
+      if (!action.payload.excludeAddonBySlug) {
+        return state;
+      }
+
       return {
         ...state,
         byAddonSlug: {
           ...state.byAddonSlug,
-          [action.payload.slug]: undefined,
+          [action.payload.excludeAddonBySlug]: undefined,
         },
       };
-    case LOAD_OTHER_ADDONS_BY_AUTHORS:
+    case LOAD_ADDONS_BY_AUTHORS:
+      let newAddons = { byAddonSlug: {}, byUserId: {} };
+
+      if (action.payload.excludeAddonBySlug) {
+        newAddons.byAddonSlug = {
+          [action.payload.excludeAddonBySlug]: action.payload.addons
+            .slice(0, ADDONS_BY_AUTHORS_PAGE_SIZE)
+            .map((addon) => createInternalAddon(addon))
+        };
+      }
+
+      const addons = action.payload.addons
+        .map((addon) => createInternalAddon(addon));
+
+      for (const addon of addons) {
+        if (addon.authors) {
```

Comment:
Flow required this check, even though I didn't think an addon could have undefined authors in this case. Maybe I'm missing something.

Reply:
I believe that's based on the types, so wherever the type is defined for `addon` it must say that `authors` is optional/can be null.

Reply:
> Flow required this check, even though I didn't think an addon could have undefined authors in this case. 

It's possible that our `ExternalAddonType` is wrong (or maybe it's trying to account for themes?). The [docs](https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#detail) say that `authors` will always be an array but you may want to double check that on the real site before adjusting the type.

It might be a bit more concise to protect against empty `authors` like:

```js
for (const author of addon.authors || []) {
}
```

Reply:
It's more concise but seems a bit less obvious to read. I feel like my `if` statement, while more verbose, is a bit more clear in setting expectation.